app/public.js

```diff
@@ -309,8 +313,11 @@
     }
   }
 
+  // video has currentSrc which points to the video file so we need to ignore it
   function getUrl(element) {
-    if (element.currentSrc) {
+    if (element.poster) {
+      return element.poster;
+    } else if (element.tagName.toLowerCase() !== 'video' && element.currentSrc) {
```

Comment:
@tabanliviu KISS - This indicates that the previous condition isn't reliable enough, so we're pushing the decision elsewhere.  Perhaps it makes more sense to be explicit and inclusive? Are there any other situations where simply checking for presence of currentSrc may not be suitable?

```
if (element.tagName.toLowerCase() === 'video') {
  return !!element.poster;
}
else if (element.currentSrc) {
   ...
}
else {
  ...
}
```